credentials/apps/badges/signals/handlers.py

```diff
@@ -51,9 +52,11 @@ def handle_badge_completion(sender, username, badge_template_uuid, **kwargs):  #
     - badge template ID
     """
 
-    # TODO: issue UserCredential
-    # TODO: ussue Credly badge
-    notify_badge_awarded(username, badge_template_uuid)
+    credential = BadgeTemplateIssuer.issued_credential_type.objects.get(uuid=badge_template_uuid)
+    BadgeTemplateIssuer().issue_credential(
+        credential,
+        username
+    )
```

Comment:
I'm looking here for something like this:

```python
CredlyBadgeTemplateIssuer().issue(badge_template_id, username)
```
- so, BADGE_PROGRESS_COMPLETE signal is refactored from `uuid` to `id` usage;
- credential lookup (by `id`) is now an inner issuer responsibility (e.g. `_get_credential(credential_id)`;
- separate explicit issuer methods for issuing/revocation;

Reply:
updated